assets/src/components/mapPage.tsx

```diff
@@ -57,70 +65,298 @@ const ToggleMobileDisplayButton = ({
+// #region Map Display
+const useCachedVehicleById = (
+  selectedVehicleOrGhost: VehicleOrGhost | null,
+  selectedVehicleId: string | null
+) => {
+  const ref = useRef(
+    (selectedVehicleOrGhost &&
+      isVehicle(selectedVehicleOrGhost) &&
+      selectedVehicleOrGhost) ||
+      null
   )
-  const selectedVehicleShapes = useTripShape(liveVehicle?.tripId || null)
 
-  const onSearchCallback = () => {
-    setSelectedVehicleId(null)
+  if (
+    selectedVehicleId === null ||
+    (selectedVehicleOrGhost && isGhost(selectedVehicleOrGhost))
+  ) {
+    ref.current = null
+  } else if (selectedVehicleOrGhost !== null) {
+    ref.current = selectedVehicleOrGhost
   }
+  return ref.current
+}
+
+const useVehicleForIdCached = (
+  socket: Socket | undefined,
+  selectedVehicleId: string | null
+) => {
+  const selectedVehicleOrGhost =
+    useVehicleForId(socket, selectedVehicleId ?? null) || null
+
+  const selectedVehicleRef = useCachedVehicleById(
+    selectedVehicleOrGhost,
+    selectedVehicleId
+  )
+  return selectedVehicleRef
+}
+
+const RouteVehicles = ({
+  selectedVehicleRoute,
+  selectedVehicleId,
+  onPrimaryVehicleSelect,
+}: {
+  selectedVehicleId: VehicleId | null
+  selectedVehicleRoute: RouteId | null
+  onPrimaryVehicleSelect: (vehicle: Vehicle) => void
+}) => {
+  const { socket } = useSocket()
+  const vehicles = useVehiclesForRoute(socket, selectedVehicleRoute)
+  return (
+    <>
+      {filterVehicles(vehicles).map((vehicle: Vehicle) => (
+        <VehicleMarker
+          key={vehicle.id}
+          vehicle={vehicle}
+          isPrimary={true}
+          isSelected={vehicle.id === selectedVehicleId}
+          onSelect={onPrimaryVehicleSelect}
+        />
+      ))}
+    </>
+  )
+}
+
+const onFollowerUpdate: UpdateMapFromPointsFn = (map, points) => {
+  if (points.length === 0) {
+    // If there are no points, blink to default center
+    map.setView(defaultCenter, 13, { animate: false })
+    return
+  }
+
+  const { width, height } = map.getContainer().getBoundingClientRect()
+  const mapContainerBounds = new Bounds([0, 0], [width, height])
+
+  // ```
+  // vpcElement.getBoundingClientRect().right - mapElement.getBoundingClientRect().left
+  //  -> 445
+  // ```
+  // Create a new inner bounds from the map bounds + "padding" to shrink the
+  // inner bounds
+  // In this case, we get the top left of the inner bounds by padding the left
+  // with the distance from the right side of the VPC to the left side of the
+  // map container
+  const topLeft = new Point(445, 0)
+  const innerBounds = new Bounds(topLeft, mapContainerBounds.getBottomRight())
+  // The "new center" is the offset between the two bounding boxes centers
+  const offset = innerBounds
+    .getCenter()
+    .subtract(mapContainerBounds.getCenter())
+
+  const targetZoom = 16
+  const targetPoint = map
+      // Project the target point into screenspace for the target zoom
+      .project(points[0], targetZoom)
```

Comment:
Because it was easier to implement for MVP heh.
The logic to fit a bounds would be similar but need a bit more thinking/work, so I left that to the future.

I wasn't sure how to allow the follower to change the type of it's `positions` prop based on the `onUpdate` function at the time of implementaiton, but that could probably be done?(unsure how long it'd take)
But then the component should be able to take a `LatLng | null` instead of `LatLng[]` if the component could change the type of the `positions` prop based on the second argument type of the `onUpdate` prop.

Reply:
I just had it come to me that I was wrong about implementation difficulty, I'll make a patch for review!

Reply:
Ohh maybe I missed something about this the first time around - is points is only ever a list of size one? I had it in my head that it could be a list including all the visible vehicles

Reply:
Yep! https://github.com/mbta/skate/pull/1891/files#diff-a6a64d5b0e8199bb4a7ea39da345d5247083943b92fd078e359cb4d1d9ab60ffR259